stdlib/ballerina-http/src/main/java/org/ballerinalang/net/uri/parser/Node.java

```diff
@@ -59,54 +60,59 @@ public DataElementType getDataElement() {
         return node;
     }
 
-    public DataElementType matchAll(String uriFragment, Map<String, String> variables, int start) {
+    public boolean matchAll(String uriFragment, Map<String, String> variables,
+                                                          int start, InboundMsgType inboundMsg,
+                                                          DataReturnAgent<DataType> dataReturnAgent) {
         int matchLength = match(uriFragment, variables);
         if (matchLength < 0) {
-            return null;
+            return false;
         }
         if (matchLength == uriFragment.length()) {
-            return dataElement;
+            return dataElement.getData(inboundMsg, dataReturnAgent);
         }
         if (matchLength >= uriFragment.length()) {
-            return null;
+            return false;
```

Comment:
Shall we improve this logic. the = part of matchLength >= uriFragment.length() condition is already satisfied by the matchLength == uriFragment.length(). So this logic is misleading. So shall we re-factor this to something like below which is more clear 
```
int matchLength = match(uriFragment, variables);
int fragmentLength = uriFragment.length();
if (matchLength < 0 || matchLength > fragmentLength ) {
            return false;
        }
        if (matchLength == fragmentLength) {
            return dataElement.getData(inboundMsg, dataReturnAgent);
        }
```